src/visualization/associations/a_priori_vis.py

```diff
@@ -0,0 +1,163 @@
+            self.left_box.setMinimum(1)
+            self.right_box = QSpinBox()
+            self.right_box.setMinimum(1)
+            self.left_button = QPushButton("PREV")
+            self.left_button.clicked.connect(partial(self.click_listener, 'prev'))
+            self.right_button = QPushButton("NEXT")
+            self.right_button.clicked.connect(partial(self.click_listener, 'next'))
+            self.step_label = QLabel("STEP: {}".format(self.current_step))
+            self.control_buttons_layout.addWidget(self.left_button)
+            self.control_buttons_layout.addWidget(self.left_box)
+            self.control_buttons_layout.addStretch()
+            self.control_buttons_layout.addWidget(self.step_label)
+            self.control_buttons_layout.addStretch()
+            self.control_buttons_layout.addWidget(self.right_box)
+            self.control_buttons_layout.addWidget(self.right_button)
+
+            self.visualization_box_layout.addLayout(self.control_buttons_layout, 0)
+
+    def _render_description(self):
+        description = "A-priori algorithm - steps visualization"
+
+        self.description_label = QLabel(description)
+        self.description_label.setWordWrap(False)
+        self.description_label.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+
+        self.description_group_box = QGroupBox()
+        self.description_group_box.setTitle("Description")
+        self.description_group_box_layout = QVBoxLayout(self.description_group_box)
+
+        self.scroll_box = QGroupBox()
+        self.scroll_box_layout = QFormLayout(self.scroll_box)
+        self.scroll = QScrollArea()
+        self.scroll.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.scroll.setVerticalScrollBarPolicy(Qt.ScrollBarAsNeeded)
+        self.scroll.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+        self.scroll.setWidget(self.scroll_box)
+        self.scroll.setWidgetResizable(True)
+        self.scroll.setMinimumHeight(26)
+        self.description_group_box_layout.addWidget(self.scroll)
+
+        self.scroll_box_layout.addWidget(self.description_label)
+        return self.description_group_box
+
+    def click_listener(self, button_type: str):
+        match button_type:
+            case 'prev':
+                self.change_step(-self.left_box.value())
+            case 'next':
+                self.change_step(self.right_box.value())
+            case 'restart':
+                self.animation = None
+                self.change_step(-self.current_step)
+                self.change_enabled_buttons(True)
+                self.interval_box.setEnabled(True)
+                self.run_button.setEnabled(True)
+            case 'run':
+                self.is_running = not self.is_running
+                self.change_enabled_buttons(False)
+                self.interval_box.setEnabled(False)
+                self.restart_button.setEnabled(not self.is_running)
+
```

Comment:
```suggestion
                self.restart_button.setEnabled(not self.is_running)
                if self.is_running:
                    self.run_button.setText("Stop animation")
                else:
                    self.run_button.setText("Start animation")

```
I understand that animation is not implemented yet, but it is important part of template of animation code.